File: snippets/snippets/node/chat-v2-post/stream-documents.js
const { CohereClientV2 } = require('cohere-ai');

const cohere = new CohereClientV2({});

(async () => {
  const stream = await cohere.chatStream({
    model: 'command-a-03-2025',
    documents: [
      {
        data: {
          text: 'Cohere is the best!',
        },
      },
    ],
    messages: [
      {
        role: 'user',
        content: [{ type: 'text', text: "Who's the best?" }],
      },
    ],
  });

  for await (const chatEvent of stream) {
    console.log(chatEvent);
  }
})();

File: snippets/snippets/python/chat-v2-post/stream-documents.py
import cohere

co = cohere.ClientV2()

response = co.chat_stream(
    model="command-a-03-2025",
    documents=[
        {"id": "1", "data": {"text": "Cohere is the best!", "title": "The best"}}
    ],
    messages=[{"role": "user", "content": "Who's the best?"}],
)

for event in response:
    if event.type == "message-start":
        print("\nMessage started.")
    elif event.type == "message-end":
        print("\nMessage ended.")
    elif event.type == "content-delta":
        print(event.delta.message.content.text, end="")

